api/controllers/search.js

```diff
@@ -280,21 +280,93 @@ var searchCollection = async function (roles, keywords, collection, pageNum, pag
   }
 
   if (collection === 'Project') {
-    // pop proponent if exists.
-    aggregation.push(
-      {
-        '$lookup': {
-          "from": "epic",
-          "localField": "proponent",
-          "foreignField": "_id",
-          "as": "proponent"
-        }
+    var projectDataId;
+
+    switch (projectLegislation) {
+      case "1996":
+        projectDataId = "allProjectData[1996]";
+        break;
+      case "2002":
+        projectDataId = "allProjectData[2002]";
+        break;
+      case "2018":
+        projectDataId = "allProjectData[2018]";
+        break;
+      case "all":
+        projectDataId = [ "allProjectData[1996]", "allProjectData[2002]", "allProjectData[2018]" ];
```

Comment:
Combine this logic with the aggregation  and then we can use `allProjectData[projectLegislation]` for the previous cases in this switch to clean up this logic. Looks good though